lib/date_time/date_time.c

```diff
@@ -64,8 +80,12 @@ static struct sntp_time sntp_time;
 K_SEM_DEFINE(time_fetch_sem, 0, 1);
 
 static struct k_work_delayable time_work;
+#if defined(CONFIG_DATE_TIME_AUTO)
+static struct k_work init_auto_work;
+#endif
 
 static struct time_aux {
+	/* TODO: Should we remove date_time_utc and rely on OS time? */
 	int64_t date_time_utc;
 	int64_t last_date_time_update;
 } time_aux;
```

Comment:
While I'd like to remove this structure and use Zephyr's time that we would return in date_time_now() etc., if we want to complete this next week, I'm not sure if we get enough validation for that quickly. So one option is also to leave it for now and do this first step to update OS time automatically, and after a month or so do more clean up.
Or I don't know if there is a reason to keep these even though we set the time to OS?
@simensrostad Let me know your thoughts on this.

Reply:
Sounds like a good idea to me to try to get rid of this.

One more thing which I think we should consider is whether modem would be more accurate in keeping track of the time. This is possible, I think it's even probable. I can do a test run during the weekend by disabling network time updates from the modem and then set the same time to both modem and application side and see what happens in a day or two.

Reply:
I'll separate the discussion about using modem time to a separate conversation thread.
Let's keep this thread on the topic of whether to keep track of the clock in date_time lib or rely on something else. And have separate conversation on that that "something else" would be.

Reply:
Managing time via the OS sounds like the better choice. That would also accommodate simulation, I believe.

Reply:
_and after a month or so do more clean up._ On that promise, yes.

Reply:
If I remove `time_aux`, I think we need to always store the time to Zephyr and not do it based on `CONFIG_DATE_TIME_SYSTEM_UPDATE` flag. Otherwise we couldn't return the time without having this own bookkeeping as `clock_gettime()` would return year 1970 etc.

Reply:
Even if we remove time_aux, we need to track previous something like `last_date_time_update` in order to handle the periodic updates.
Right now I'm not sure we get this done before Thursday.